crates/ra_hir_def/src/path/lower.rs

```diff
@@ -113,6 +116,20 @@ pub(super) fn lower_path(mut path: ast::Path, hygiene: &Hygiene) -> Option<Path>
     }
     segments.reverse();
     generic_args.reverse();
+
+    // handle local_inner_macros :
+    // Basically, even in rustc it is quite hacky:
+    // https://github.com/rust-lang/rust/blob/614f273e9388ddd7804d5cbc80b8865068a3744e/src/librustc_resolve/macros.rs#L456
+    // We follow what it did anyway :)
+    if segments.len() == 1 && kind == PathKind::Plain {
+        let next = path.syntax().last_token().and_then(|it| it.next_token());
+        if next.map_or(false, |it| it.kind() == T![!]) {
```

Comment:
Hmm, this "followed-by-!" logic seems particularly hacky, and isn't what rustc is doing.

I don't really see how this could break anything though, so I suppose it's fine.

Reply:
Yeah, let change it to test the parent node whether it is a macro call and bang macro like exactly what rustc does.